src/Bicep.LangServer/Completions/BicepCompletionContext.cs

```diff
@@ -154,9 +154,9 @@ private static List<SyntaxBase> FindNodesMatchingOffset(ProgramSyntax syntax, in
         private static BicepCompletionContextKind GetDeclarationTypeFlags(IList<SyntaxBase> matchingNodes, int offset)
         {
             // local function
-            bool IsStuff(SyntaxBase name, SyntaxBase type) => name.Span.Length > 0 && type.Span.Position == offset;
+            bool CheckTypeIsExpected(SyntaxBase name, SyntaxBase type) => name.Span.Length > 0 && offset > name.Span.Position + name.Span.Length && offset <= type.Span.Position;
```

Comment:
Worth making this into a helper in `IPositionableExtensions`? Feels like it could be useful elsewhere

Reply:
Also, I love the name `IsStuff` :) I do that too when I can't be bothered to name something immediately.

Reply:
Yeah that's a good pt. It comes up once in a while, so it'll be useful. I'll add an extension method. (Yeah was debating if I should rebase to get rid of IsStuff, but decided to leave it hehe.)